pkgs/development/python-modules/dtfabric/default.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pytest-mock,
+  pytestCheckHook,
+  pyyaml,
+  setuptools,
+}:
+buildPythonPackage rec {
+  pname = "dtfabric";
+  version = "20230520";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-rJPBEe/eAQ7OPPZHeFbomkb8ca3WTheDhs/ic6GohVM=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    pyyaml
+  ];
+
+  nativeCheckInputs = [
+    pytest-mock
+    pytestCheckHook
+  ];
+
+  pythonRemoveDeps = [
+    "pip"
+  ];
+
+  preCheck = ''
+    patchShebangs run_tests.py
+  '';
+
+  checkPhase = ''
+    python3 run_tests.py
+  '';
```

```suggestion
  checkPhase = ''
    runHook preCheck
    ${python.interpreter} run_tests.py
    runHook postCheck
  '';
```
Please add `runHook` and use `${python.interpreter}` to enable cross-building.